smartsim/entity/model.py

```diff
@@ -291,7 +303,39 @@ def _set_colocated_db_settings(
         if hasattr(self.run_settings, "_prep_colocated_db"):
             self.run_settings._prep_colocated_db(common_options["cpus"])
 
+        if "limit_app_cpus" in common_options:
+            raise SSUnsupportedError(
+                "Pinning of app CPUs via limit_app_cpus is no supported. Modify RunSettings " +
+                "instead using the correct binding option for your launcher."
+            )
+
         # TODO list which db settings can be extras
+        cpus = common_options["cpus"]
+        # Deal with cases where the database should be pinned to cpus
+        # (1) if the user set a db_cpu_list, but not limit_db_cpus
+        if common_options["db_cpu_list"] and not common_options["limit_db_cpus"]:
+            logger.warning(
+                "limit_db_cpus is False, but db_cpu_list is not None. Setting limit_db_cpus=True"
+            )
+            common_options["limit_db_cpus"] = True
```

Comment:
^^ Really seems like we can drop `limit_db_cpus` and just assume that if `db_cpus_list is not None` that we should limit to those cpus

Reply:
Yah so this is where I disagree, I think there's only one unallowed pathway which we should make sure that the users are aware of and fail fast on, which is `limit_db_cpus=False` and `db_cpus_list` is something.

Reply:
Will followup offline